src/lib/logSummary/logSummary.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <unistd.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <pthread.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "alarmMgr/alarmMgr.h"
+#include "logSummary/logSummary.h"
+
+
+
+/* ****************************************************************************
+*
+* Global vars
+*/
+static sem_t               lsSem;
+static unsigned long long  transactionCounter           = 0;
+static unsigned long long  transactionsSinceLastSummary = 0;
+static bool                logSummaryOn                 = false;
+
+
+
+/* ****************************************************************************
+*
+* transactionCounterIncrement - 
+*/
+void transactionCounterIncrement(void)
+{
+  if (logSummaryOn == false)
+  {
+    return;
+  }
+
+  sem_wait(&lsSem);
+  ++transactionCounter;
+  ++transactionsSinceLastSummary;
+  sem_post(&lsSem);
```

Comment:
I would propose a different strategy:

* logSummary keeps only an internal counter, named `transactionCounterForSummary` (or a better name) initialized to 0
* each time the thread wakes up to print, you get the current value of `transaction` (I refer to the transaction counter, a static int defined in transactionIdSet). Given that this is a read operation, you don't need to wait on transaction semaphoere (i.e. the one managed with transSemTake/transSemGive) for this.
* You use for the trace log the difference `transaction - transactionCounterForSummary`
* Update the `transactionCounterForSummary` with `transaction` for the next round
* Sleep until next wake up.

I think this will lead to a simpler code, as there is no need of using transactionCounterIncrement() from rest.cpp and httpRequestSend.cpp

In addition, probably there isn't a need for lsSem, as no thread (appart from the one managing log summary, of course) will touch `transactionCounterForSummary`.

(Atually, is not so simple in step 3... the difference could be negative, in the case of overflow, but it is easy to adjust in that case).